examples/bri-3/src/bri/workgroup/workgroups/api/workgroups.controller.spec.ts

```diff
@@ -178,6 +184,49 @@ describe('WorkgroupsController', () => {
     });
   });
 
+  describe('archiveWorkgroup', () => {
+    it('should throw NotFound if non existent id passed', async () => {
+      // Arrange
+      const nonExistentId = '123';
+
+      // Act and assert
+      expect(async () => {
+        await workgroupController.archiveWorkgroup(nonExistentId);
+      }).rejects.toThrow(
+        new NotFoundException(WORKGROUP_NOT_FOUND_ERR_MESSAGE),
+      );
+    });
+
+    it('should throw BadRequest if workgroupToArchive status is not active', async () => {
+      // Arrange
+      const newWorkgroupId = await createTestWorkgroup();
+
+      await workgroupController.archiveWorkgroup(newWorkgroupId);
+
+      // Act and assert
+      expect(async () => {
+        await workgroupController.archiveWorkgroup(newWorkgroupId);
+      }).rejects.toThrow(
+        new BadRequestException(WORKGROUP_STATUS_NOT_ACTIVE_ERR_MESSAGE),
+      );
+    });
+
+    it('should perform the update if existing id passed', async () => {
+      // Arrange
+      const newWorkgroupId = await createTestWorkgroup();
+
+      // Act
+      await workgroupController.archiveWorkgroup(newWorkgroupId);
+
+      // Assert
+      const archivedWorkgroup = await workgroupController.getWorkgroupById(
+        newWorkgroupId,
+      );
+      expect(archivedWorkgroup.id).toEqual(newWorkgroupId);
+      expect(archivedWorkgroup.status).toEqual(WorkgroupStatus.ARCHIVED);
```

Comment:
I could see a world where other workgroup values are checked to ensure they did not change, but I am not gonig to die on that hill.

Reply:
good point, can be done but also not dying on the hill

Reply:
I will add check for all the values in the next PR related to this branch.